src/middleware/TestingHandler.ts

```diff
@@ -0,0 +1,255 @@
+	 * @param {string} requestMethod - the API method for the request
+	 * @returns Response object
+	 */
+	private setStatusCode(testingHandlerOptions: TestingHandlerOptions, requestURL: string, requestMethod: string): Response {
+		try {
+			testingHandlerOptions.statusMessage = "Some error happened";
+			if (testingHandlerOptions.testingStrategy === TestingStrategy.MANUAL) {
+				if (testingHandlerOptions.statusCode === undefined) {
+					try {
+						const urlMethod = this.getRelativeURL(new RegExp("http(s)://graph.microsoft.com/[^/]*", "g"), requestURL);
+						testingHandlerOptions.statusCode = this.manualMap.get(urlMethod).get(requestMethod);
+					} catch (error) {
+						const urlMethod = this.getRelativeURL(new RegExp("http(s)://graph.microsoft.com/[^/]*", "g"), requestURL);
+						this.manualMap.forEach((value: Map<string, number>, key: string) => {
+							const regexUrl = new RegExp(key);
+							if (regexUrl.test(urlMethod)) {
+								testingHandlerOptions.statusCode = this.manualMap.get(key).get(requestMethod);
+							}
+						});
+
+						if (testingHandlerOptions.statusCode === undefined) {
+							if (requestURL === this.redirectURL) {
+								testingHandlerOptions.statusCode = 404;
+							} else {
+								throw new Error("API not available in map");
+							}
+						}
+					}
+				} else {
+					if (requestURL === this.redirectURL) {
+						const statusCode: number = testingHandlerOptions.statusCode;
+						if (statusCode === 301 || statusCode === 302 || statusCode === 303 || statusCode === 307 || statusCode === 308) {
+							testingHandlerOptions.statusCode = 404;
+						}
+					}
+				}
+			} else if (testingHandlerOptions.testingStrategy === TestingStrategy.RANDOM) {
+				testingHandlerOptions.statusCode = this.getStatusCode(requestMethod);
+			}
+
+			return this.createResponse(testingHandlerOptions, requestURL);
+		} catch (error) {
+			throw error;
+		}
+	}
+
+	/**
+	 * @private
+	 * To get the options for execution of the middleware
+	 * @param {Context} context - The context object
+	 * @returns A options for middleware execution
+	 */
+	private getOptions(context: Context): TestingHandlerOptions {
+		let options: TestingHandlerOptions;
+		if (context.middlewareControl instanceof MiddlewareControl) {
+			options = context.middlewareControl.getMiddlewareOptions(TestingHandlerOptions) as TestingHandlerOptions;
+		}
+		if (typeof options === "undefined") {
+			options = Object.assign(new TestingHandlerOptions(), this.options);
+		}
+		return options;
+	}
+
+	/**
+	 * @public
+	 * @async
+	 * To execute the current middleware
+	 * @param {Context} context - The context object of the request
+	 * @returns A Promise that resolves to nothing
+	 */
+	public async execute(context: Context): Promise<void> {
+		try {
+			const testingHandlerOptions = this.getOptions(context);
+			context.response = this.setStatusCode(testingHandlerOptions, context.request as string, context.options.method as string);
```

Comment:
I suggest that we alter createResponse to take an HttpMethod, so that we could call this.createResponse to get a Response object instead of calling this.setStatusCode to get a Response object. 

Then have createResponse call setStatusCode before forming the Response object.

Reply:
Please provide a response about why this should stay as is before resolving the conversation with no action.

Reply:
Ugh, was looking at out dated version. All good.